quinn/src/endpoint.rs

```diff
@@ -69,13 +68,12 @@ where
         addr: &SocketAddr,
         server_name: &str,
     ) -> Result<Connecting<S>, ConnectError> {
-        self.connect_with(
-            self.default_client_config
-                .get_or_init(ClientConfig::default)
-                .clone(),
-            addr,
-            server_name,
-        )
+        let config = match &self.default_client_config {
+            Some(config) => config.clone(),
+            None => return Err(ConnectError::NoDefaultClientConfig),
```

Comment:
Could the introduction of this error, and removal of `OnceCell`, be factored out into a separate commit? Also, can/should the `once_cell` dep be dropped?

Reply:
Done. Had some sloppy rebasing going on that nixed the removal of the dependency.